core/modules/query/QueryContext.cc

```diff
@@ -142,6 +142,18 @@ QueryContext::getValueExprMatch(std::shared_ptr<query::ValueExpr> const& valExpr
 }
 
 
+void QueryContext::addRestrictors(RestrList const& restrictors_) {
+    if (restrictors_.empty()) {
+        return;
+    }
+    if (nullptr == restrictors) {
```

Comment:
This test is really confusing, primarily because the type `RestrList` of the input parameter has no indication that it's the shared pointer type. Could be renamed into `RestrListPtr` perhaps?

Reply:
I'm disinclined to rename a type to `...Ptr` this without a broader effort (at least in the Czar code); there's a lot of places where containers are owned by a shared_ptr and _don't_ follow that convention, and to mix & match, at least not without an explicit intention of going one way or the other (and I've been admonished for trying to introduce Hungarian Notation before...) seems like it's introducing more confusion than clarity. OTOH, there are 2 variables here: `restrictors` and `restrictors_`, the latter has that underscore which we often use to mean "input variable" (grumble). I'll rename the input var to `newRestrictors` or something, to make it more clear.